src/librustc/ty/maps/plumbing.rs

```diff
@@ -583,7 +595,7 @@ macro_rules! define_maps {
 
 macro_rules! define_map_struct {
     (tcx: $tcx:tt,
-     input: ($(([$(modifiers:tt)*] [$($attr:tt)*] [$name:ident]))*)) => {
```

Comment:
the problem was here. This should have been:

```
$($modifiers:tt)*
```

note the `$` before modifiers

Reply:
$($())*